lib/license_scout/dependency_manager/cpan.rb

```diff
@@ -0,0 +1,307 @@
+          FileUtils.mkdir_p(license_cache_root)
+          FileUtils.cp(unpacked_file, license_cache_path)
+          license_cache_path
+        end
+
+        def slurp_meta_yaml_in(unpack_path)
+          metadata = Psych.safe_load(File.read(meta_yaml_in(unpack_path)))
+          if metadata.key?("license")
+            given_type = Array(metadata["license"]).reject {|l| l == "unknown" }.first
+            @license = normalize_license_type(given_type)
+          end
+        end
+
+        def slurp_meta_json_in(unpack_path)
+          metadata = FFI_Yajl::Parser.parse(File.read(meta_json_in(unpack_path)))
+          if metadata.key?("license")
+            given_type = Array(metadata["license"]).reject {|l| l == "unknown" }.first
+            @license = normalize_license_type(given_type)
+          end
+        end
+
+        def license_cache_root
+          File.join(cache_root, "cpan-licenses")
+        end
+
+        def dist_cache_root
+          File.join(cache_root, "cpan-dists")
+        end
+
+        def normalize_license_type(given_type)
+          LICENSE_TYPE_MAP[given_type] || given_type
+        end
+
+        def meta_json_in(unpack_path)
+          File.join(unpack_path, "META.json")
+        end
+
+        def mymeta_json_in(unpack_path)
+          File.join(unpack_path, "MYMETA.json")
+        end
+
+        def meta_yaml_in(unpack_path)
+          File.join(unpack_path, "META.yml")
+        end
+
+        def found_license_files_in(unpack_path)
+          Dir["#{unpack_path}/*"].select do |f|
+            CPAN::POSSIBLE_LICENSE_FILES.include?(File.basename(f))
+          end
+        end
+
+      end
+
+
+      def initialize(*args, &block)
+        super
+        @dependency_graph_xml_file = nil
+        @dependencies = nil
+      end
+
+      def name
+        "perl_cpan"
+      end
+
+      def dependencies
+        return @dependencies if @dependencies
+        @dependencies = deps_list.map do |d|
+          d.collect_licenses
+          d.to_dep
+        end
+      end
+
+      def deps_list
+        return @deps_list if @deps_list
+
+        xml_doc = REXML::Document.new(dependency_graph_xml)
+
+        root = xml_doc.root
+
+        deps = root.get_elements("//dependency")
+
+        @deps_list = []
+
+        deps.each do |dep|
+          dep_module_name = dep.get_text("module").to_s
+          next if dep_module_name == module_name
+          @deps_list << CPANDependency.new(
+            module_name: dep_module_name,
+            dist: dep.get_text("dist").to_s,
+            version: dep.get_text("distversion").to_s,
+            cpanfile: dep.get_text("cpanfile").to_s,
+            cache_root: options.cpan_cache
+          )
+        end
+
+        @deps_list
+      end
+
+      def dependency_graph_xml
+        @dependency_graph_xml ||= File.read(dependency_graph_xml_file)
+      end
+
+      def dependency_graph_xml_file
+        @dependency_graph_xml_file ||= NetFetcher.cache(dependency_graph_url)
```

Comment:
I think the licenses for specific dists should be okay; even if we get a dependency upgrade, the logic should result in us overwriting the old cached license file. I'll look into cleaning up the XML file.